doc/source/user/c-info.ufunc-tutorial.rst

```diff
@@ -193,6 +193,58 @@ site-packages directory.
 
         setup(name='spam', version='1.0', ext_modules=[module1])
 
+.. note::
+
+   For modern Python packaging, it is recommended to use a `pyproject.toml` and select
+   `meson-python` as your build backend. This makes your extension pip-installable and
+   compatible with current Python packaging standards.
+
+   Example `pyproject.toml`:
+
+   .. code-block:: toml
+
+      [build-system]
+      requires = ["meson-python", "numpy"]
+      build-backend = "mesonpy"
+
+.. tab-set::
+
+   .. tab-item:: meson
+
+      .. code-block:: meson
+
+         project('spam', 'c')
+
+         py = import('python').find_installation()
+         numpy_include = run_command(py.full_path(), '-c', 'import numpy; print(numpy.get_include())').stdout().strip()
+
+         sources = files('spammodule.c')
+
+         extension_module = py.extension_module(
+           'spam',
+           sources,
+           include_directories: [numpy_include],
+           install: true
```

Comment:
Let's add `subdir: 'pkgname',` to this, which makes the extension module land in the right place.